src/cargo/util/config/de.rs

```diff
@@ -195,11 +195,35 @@ impl<'de, 'config> de::Deserializer<'de> for Deserializer<'config> {
         }
     }
 
+    fn deserialize_enum<V>(
+        self,
+        name: &'static str,
+        _variants: &'static [&'static str],
+        visitor: V,
+    ) -> Result<V::Value, Self::Error>
+    where
+        V: de::Visitor<'de>,
+    {
+        if name == "Strip" {
```

Comment:
I don't think it is necessary to look at the name here.  If the enum does not support deserializing from a simple string, then the call to `visit_enum` will fail and return an appropriate error.